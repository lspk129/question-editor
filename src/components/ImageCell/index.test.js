import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ImageCell } from './';
import imageIcon from '../../icons/ic_image_black_24px.svg';

describe('ImageCell', () => {
  const props = {
    change: () => {},
    id: '12345',
    icon: imageIcon,
  };
  const wrapper = mount(
    <ImageCell {...props} />,
  );

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has change prop', () => {
    expect(wrapper).toHaveProp('change');
  });

  it('has with id prop', () => {
    expect(wrapper).toHaveProp('id', '12345');
  });

  it('has icon props', () => {
    expect(wrapper).toHaveProp('icon', imageIcon);
  });
});
