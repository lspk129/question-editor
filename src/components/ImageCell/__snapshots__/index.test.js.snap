// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCell renders correctly 1`] = `
<Component
  change={[Function]}
  icon="ic_image_black_24px.svg"
  id="12345"
>
  <styled.label
    icon="ic_image_black_24px.svg"
  >
    <label
      className="sc-bdVaJa cdXfjg"
      icon="ic_image_black_24px.svg"
    >
      <styled.input
        data-img="12345"
        onChange={[Function]}
        type="file"
      >
        <input
          className="sc-bwzfXH eKeXgs"
          data-img="12345"
          onChange={[Function]}
          type="file"
        />
      </styled.input>
    </label>
  </styled.label>
</Component>
`;
